### PR TITLE
path-win: implement cache directories

### LINKED ISSUE #11898: vo=gpu/gpu-next: cache files are generated in config directory after mpv closes
### Important Information

- mpv version: [MPV 2023-07-07 13:04](https://github.com/zhongfly/mpv-winbuild/releases/tag/2023-07-07)
- Platform and Version: Windows 11 x64
- Source of the mpv binary: [mpv-x86_64-v3-20230707-git-6ed521f.7z](https://github.com/zhongfly/mpv-winbuild/releases/download/2023-07-07/mpv-x86_64-v3-20230707-git-6ed521f.7z)

### Reproduction steps

1. Start with an empty/default configuration.
2. Start mpv by `mpv --vo=gpu-next` (Note: no need to play any video file here; just start the gui.)
3. Close mpv.

### Expected behavior

No extra file should be generated.

### Actual behavior

A file called `libplacebo.cache` is generated in the configuration folder. In my case, it's generated under `portable_config`.

Is this a bug or a new expected behavior in the latest version? If it's the latter, is there a way to disable it?

### Log file

[output.txt](https://github.com/mpv-player/mpv/files/11985075/output.txt)

### Sample files

N/A as this issue does not need a video file to reproduce. On the other hand, a sample `libplacebo.cache` file is attached below. (I zipped it as GitHub doesn't allow uploading a raw `.cache` file.)

[libplacebo.cache.zip](https://github.com/mpv-player/mpv/files/11985091/libplacebo.cache.zip)

### PR DESCRIPTION
**Pull request by Dudemanguy**
First, someone please make sure this actually works since I didn't test it at all. This implements separate cache directories in windows. It's meant to be `%LOCALAPPDATA%`. It seems like everyone agrees on that one at least. For the portable_config path, it should be a separate subdirectory within it called `cache`. Dunno how people feel about that. Feel free to complain. Fixes #11898.

### CODE REVIEW

**Comment by hooke007** [comment]
For `portable_config`, would it better to add some tips in Doc?
https://github.com/mpv-player/mpv/issues/8616
It seems many windows users do really don't know which folders usually is not writable for non-admin.

**Comment by Dudemanguy** [comment]
I could clarify that cache files are also written to the directory and that write permissions are needed I suppose.

**Comment by hooke007** [comment]
Actually I mean the unwritable dir may cause the serious performance problem.

File: DOCS/man/mpv.rst [hunk 1]
```diff
@@ -1631,16 +1651,18 @@ You can find the exact path by running ``echo %APPDATA%\mpv\mpv.conf`` in cmd.ex
 Other config files (such as ``input.conf``) are in the same directory. See the
 `FILES`_ section above.
 
+Cache fles are located under ``%LOCALAPPDATA%/mpv``.
```
Comment (sfan5) [inline DOCS/man/mpv.rst:1654]:
```suggestion
The cache directory is located at ``%LOCALAPPDATA%/mpv``.
```

File: DOCS/man/mpv.rst [hunk 2]
```diff
@@ -1539,13 +1539,26 @@ input command can take an exit code: in this case, that exit code is returned.
 FILES
 =====
 
+Note, these file paths assume Linux/BSD. On other platforms, the paths may be different.
```
Comment (sfan5) [inline DOCS/man/mpv.rst:1542]:
```suggestion
Note that this section assumes Linux/BSD. On other platforms the paths may be different.
```

File: osdep/path-win.c [hunk 3]
```diff
@@ -87,9 +95,13 @@ const char *mp_get_platform_path_win(void *talloc_ctx, const char *type)
     if (portable_path) {
         if (strcmp(type, "home") == 0)
             return portable_path;
+        if (strcmp(type, "cache") == 0)
+            return portable_cache_path;
```
Comment (sfan5) [inline osdep/path-win.c:99]: you could move the mp_path_join call here btw

**Review by sfan5 — APPROVED** [review]

**Comment by Dudemanguy** [comment]
Well I guess windows users are happy with the paths here. ¯\_(ツ)_/¯

**Pull request merged**